webpack/utils/env.js

```diff
@@ -1,17 +1,51 @@
 // @ts-check
 
+import { createHash } from "crypto"
+import { readFileSync, accessSync, constants } from "fs"
+import { parse } from "dotenv"
 import chalk from "chalk"
 import path from "path"
 import yn from "yn"
 import os from "os"
+import v8 from "v8"
+
+function loadEnv(env) {
+  return checkFileExistsSync(env) ? parse(readFileSync(env)) : {}
+}
+
+function applyToEnv(config) {
+  if (Object.keys(config).length !== 0) {
+    for (const k in config) {
+      process.env[k] = config[k]
+    }
+  }
+}
+
+function checkFileExistsSync(filepath) {
+  try {
+    accessSync(filepath, constants.F_OK)
+  } catch (e) {
+    return false
+  }
+  return true
+}
+
+// Load the default envs and apply to process.env
+applyToEnv({
```

Comment:
This looks like an improved `src/loadenv.js` implementation! Wondering why you didn't refactor that and `import src/loadenv.js` here instead?

Reply:
I debated heavily, this actually started as a refactor of `src/loadenv.js`. My preference is to keep our build source completely separate from our application logic lest it accidentally be included in a production build by webpack due to a rogue import.

It made me think twice about making a separate npm module for it 😄

Reply:
Thanks for clarifying. Let the telemetry roll in 😃